ConfigSpace/hyperparameters.pyx

```diff
@@ -877,12 +888,283 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+            name=self.name,
+            default_value=self.default_value,
+            alpha=self.alpha,
+            beta=self.beta,
+            log=self.log,
+            q=self.q,
+            lower=self.lower,
+            upper=self.upper,
+            meta=self.meta
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.alpha, self.beta, self.lower, self.upper, self.log, self.q))
+
+    def to_uniform(self) -> 'UniformFloatHyperparameter':
+        lb = self.lower
+        ub = self.upper
+
+        return UniformFloatHyperparameter(self.name,
+                                          lb,
+                                          ub,
+                                          default_value=self.default_value,
+                                          q=self.q, log=self.log)
+
+    def check_default(self, default_value: Union[int, float]) -> Union[int, float]:
+        # return mode as default
+        lb = self.lower
+        ub = self.upper
+        
+        if default_value is None:
+            if (self.alpha > 1) and (self.beta > 1):
+                normalized_mode = (self.alpha - 1) / (self.alpha + self.beta - 2)
+                return self._transform_scalar((self._upper - self._lower) * normalized_mode + self._lower)
+            else: 
+                return (ub - lb) / 2
+
+        elif self.is_legal(default_value):
+            return default_value
+        else:
+            raise ValueError("Illegal default value %s" % str(default_value))
+
+    def to_integer(self) -> 'BetaIntegerHyperparameter':
+        if self.q is None:
+            q_int = None
+        else:
+            q_int = int(np.rint(self.q))
+        
+        lower = int(np.ceil(self.lower))
+        upper = int(np.floor(self.upper))
+        default_value = int(np.rint(self.default_value))
+        return BetaIntegerHyperparameter(self.name, lower=lower, upper=upper, alpha=self.alpha, beta=self.beta,
+                                           default_value=int(np.rint(self.default_value)),
+                                           q=q_int, log=self.log)
+
+    def is_legal(self, value: Union[float]) -> bool:
+        if not (isinstance(value, float) or isinstance(value, int)):
+            return False
+        elif self.upper >= value >= self.lower:
+            return True
+        else:
+            return False
+
+    cpdef bint is_legal_vector(self, DTYPE_t value):
+        if self._upper >= value >= self._lower:
+            return True
+        else:
+            return False
+
+    def _sample(self, rs: np.random.RandomState, size: Optional[int] = None
+                ) -> Union[np.ndarray, float]:
+        alpha = self.alpha
+        beta = self.beta
+        lower = self._lower
+        upper = self._upper
+        return spbeta(alpha, beta, loc=lower, scale=upper-lower).rvs(size=size, random_state=rs)
+
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.isnan(vector).any():
```

Comment:
Don't you need to scale from the unit hypercube to the real values?

Reply:
No, as is the case with the nomalHPs. The NormalHPs are not scaled (I guess because they are potentially unbounded) and I decided to do the same with the BetaHPs. However, they can be [0, 1]-scaled just like the UniformHPs if one would prefer that route.

In general, I tried to resemble the NormalHPs as much as I could.